pkg/etcdcli/health.go

```diff
@@ -42,23 +42,24 @@ type healthCheck struct {
 type memberHealth []healthCheck
 
 func GetMemberHealth(ctx context.Context, etcdMembers []*etcdserverpb.Member) memberHealth {
-	memberHealth := memberHealth{}
-	resChan := make(chan healthCheck, 1)
-	for _, member := range etcdMembers {
+	// while we don't explicitly mention that the returned ordering has to be the same as etcdMembers,
+	// we try to keep it that way for backward compatibility reasons
+	memberHealth := make([]healthCheck, len(etcdMembers))
+	wg := sync.WaitGroup{}
+	for i, member := range etcdMembers {
 		if !HasStarted(member) {
 			memberHealth = append(memberHealth, healthCheck{Member: member, Healthy: false})
```

Comment:
Did you mean to drop the index assign for the append? This would allow `memeberHealth` to be longer than `etcdMembers` with potentially empty/nil entries?

Reply:
sigh, thank you